### PR TITLE
XQuery lexer: Unicode support

### PR DESCRIPTION
**Pull request by thvitt**
The XQuery lexer considered non-ascii characters in names and literal XML elements / attributes as errors. This pull request resolves the "fix unicode later" comments in the original source.

### CODE REVIEW

File: pygments/lexers/webmisc.py [hunk 1]
```diff
@@ -70,19 +70,17 @@ class XQueryLexer(ExtendedRegexLexer):
...
+       r"[A-Z]|_|[a-z]|[\u00C0-\u00D6]|[\u00D8-\u00F6]|[\u00F8-\u02FF]|"
+       r"[\u0370-\u037D]|[\u037F-\u1FFF]|[\u200C-\u200D]|[\u2070-\u218F]|"
+       r"[\u2C00-\u2FEF]|[\u3001-\uD7FF]|[\uF900-\uFDCF]|[\uFDF0-\uFFFD]|"
+       r"[\U00010000-\U000EFFFF]"
+    )
+    ncnamestartchar = '(?:%s)' % _ncnamestartchar
+    _ncnamechar = _ncnamestartchar + (r"|-|\.|[0-9]|\u00B7|[\u0300-\u036F]|"
+                                     r"[\u203F-\u2040]")
+    ncnamechar = "(?:%s)" % _ncnamechar
+    ncname = "(?:%s%s*)" % (ncnamestartchar, ncnamechar)
```
[the remark this answers is not in the document]
Comment (thvitt) [reply pygments/lexers/webmisc.py:83]: Your expression matches, e.g., `,x` which is not a valid NCName, but it does not match `para`, which is. I am not aware of a simple way to express _any unicode letter, but not a digit_ with plain stdlib regex's, as `\p{L}` isn't supported.

(Additionally, what `\w` matches is slightly different from [the relevant productions from the XML spec](https://www.w3.org/TR/REC-xml/#NT-NameStartChar), but that is probably only an academic difference. The ranges in the grammar are taken from the XML / XQuery specs, though)

[the remark this answers is not in the document]
Comment (thvitt) [reply pygments/lexers/webmisc.py:83]: ah, that's just one negation too much for me :) I’ll adjust it, but I’m too tired now

**Comment by ChristianGruen** [comment]
I agree it would be nice to have better Unicode support.
@thvitt Are you still working on this PR?

**Comment by thvitt** [comment]
@ChristianGruen I have rebased the old patch onto current master and also cleaned up the pitarget match. 

While trying to add a test case I noticed also non-unicode liteal processing instructions seem to be generally broken in the current version as well, I’ll have to look into this. Would you prefer a seperate issue / PR for that?

**Comment by ChristianGruen** [comment]
@thvitt Thanks, much appreciated. 

I don't have permissions for merging PRs. Maybe someone else can give some feedback?

**Comment by ChristianGruen** [comment]
Anyone (maybe @jeanas)… Could this PR be merged?

**Comment by ChristianGruen** [comment]
(maybe @Anteru)… I think about upgrading the code for newer versions of XQuery (3.1/4.0). Do you think I should wait until this PR is merged, or should I simply start a separate attempt?

**Comment by Anteru** [comment]
I think a separate PR will be better, I want to cut the release within 2 weeks (...) and then we'll revisit this.

File: pygments/lexers/webmisc.py [hunk 2]
```diff
@@ -96,20 +84,27 @@ class XQueryLexer(ExtendedRegexLexer):
...
+    # " U+0022 QUOTATION MARK
+    # & U+0026 AMPERSAND
+    # ' U+0027 APOSTROPHE
+    # < U+003C LESS-THAN SIGN
+    # > U+003E GREATER-THAN SIGN
+    # { U+007B LEFT CURLY BRACKET
+    # } U+007D RIGHT CURLY BRACKET
+    _char_rest = r'|[\u0080-\uD7FF]|[\uE000-\uFFFD]|[\U00010000-\U0010FFFF]'
+    elementcontentchar = (r'\t|\r|\n|[\u0020-\u0025]|[\u0028-\u003b]|' 
+                          r'[\u003d-\u007a]|\u007c|[\u007e-\u007F]' + _char_rest)
```
[the remark this answers is not in the document]
Comment (ChristianGruen) [reply pygments/lexers/webmisc.py:98]: In the pattern of the PR, `\u0028-\u003b` would need to be replaced with `\u0027-\u003b`.

> Wouldn't it be a lot simpler to do things like […]

I think it would then need to be…

```
elementcontentchar  = r'[^{}<&]+'
quotattrcontentchar = r'[^{}<&"]+'
aposattrcontentchar = r"[^{}<&']+"
```

**Comment by ChristianGruen** [comment]
Possibly obsolete (@thvitt I hope that’s OK) if #3259 is going to be accepted.
